Drop-In/src/androidTest/java/com/braintreepayments/api/PaymentMethodClientTest.java

```diff
@@ -0,0 +1,120 @@
+package com.braintreepayments.api;
+
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static com.braintreepayments.api.Assertions.assertIsANonce;
+import static com.braintreepayments.api.CardNumber.VISA;
+import static com.braintreepayments.api.ExpirationDate.validExpirationYear;
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+import static junit.framework.Assert.fail;
+
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class PaymentMethodClientTest {
+
+    private Context context;
+
+    @Before
+    public void setUp() {
+        context = ApplicationProvider.getApplicationContext();
+    }
+
+    @Test(timeout = 10000)
+    public void getPaymentMethodNonces_andDeletePaymentMethod_returnsCardNonce() throws InterruptedException {
```

Comment:
Should we split these up? Have 1 test that is just asserting on that getPaymentMethodNonces returns a nonce, and another test for the deletion?

Reply:
I think it makes sense for them to be seperate, but we would probably have to name the tests in a way that makes sure to run the `getPaymentMethodNonces` test before the `deletePaymentMethod` test.

Reply:
I don't think we want one test to depend on another test being run first. But we could have one test that just tests `getPaymentMethodNonces` and another one that tests `deletePaymentMethod` and calls `getPaymentMethodNonces` as part of its set-up.

Reply:
This is an odd test. The client token is generated with a completely random customerId, so I think the expectation is this test is for a new customer. With that assumption, we would have `createNonce -> assertGetNoncesWorks -> deleteNonce` and `createNonce -> deleteNonce -> assertNoNoncesArePresent`.

We would benefit from having the assertions isolated in their own tests, but the setup / teardown wouldn't be much different. I'm 50/50.

Reply:
Gotcha - also agree with Susan. Whatever you decide works for me on this one 👍

Reply:
Thank you for providing the additional info